### PR TITLE
Add person student teacher classes

### PR DESCRIPTION
**Pull request by Rayan84**
In this branch, the following has been added:

- Person class
- Student class
- Teacher class

The classes contain inherited and own methods, Person class has one private method

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission = 'true')
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (mikethreels) [inline person.rb]: - [x] according to the requirements the should also be an instance variable for `@id` in order for this to work it should be included in the `initialize` method in the same way as `@age`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    return true if of_age? || @parent_permission
+
+    false
```
Comment (mikethreels) [inline person.rb]: - [x] [OPTIONAL] including the false in this case is redundant, the method will return false without this line as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Rayan, 

Good job setting up the classes mostly as requested in the project requirements
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by oloomoses — APPROVED** [review]
Hi @Rayan84, 

Well done on making the required changes.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**